File: internal-base/utils/src/main/java/com/lijing/dev/utils/NullObjects.java
package com.lijing.dev.utils;

import java.util.Collection;

public class NullObjects {

    public static boolean isNull(Object object) {
        return object == null;
    }

    public static boolean isNotNull(Object object) {
        return !isNull(object);
    }

    public static boolean isEmpty(Object object) {
        if (isNull(object)) {
            return false;
        }

        if (object instanceof Collection) {
            return ((Collection) object).isEmpty();
        }

        if (object instanceof String) {
            return ((String) object).isEmpty();
        }

        return true;
    }

    public static boolean isNotEmpty(Object object) {
        return !isEmpty(object);
    }
}

File: internal-base/utils/src/main/java/com/lijing/dev/utils/NullObjects.kt
package com.lijing.dev.utils

object NullObjects {

    fun isNull(obj: Any?): Boolean {
        return obj == null
    }

    fun isNotNull(obj: Any): Boolean {
        return !isNull(obj)
    }

    fun isEmpty(obj: Any): Boolean {
        return if (isNull(obj)) {
            false
        } else (obj as? Collection<*>)?.isEmpty() ?: ((obj as? String)?.isEmpty() ?: true)

    }

    fun isNotEmpty(obj: Any): Boolean {
        return !isEmpty(obj)
    }
}
